examples/notebooks/beam-ml/per_key_models.ipynb

```diff
@@ -0,0 +1,597 @@
+        "\n",
+        "This notebook demonstrates how you can use a `KeyedModelHandler` to run inference in a Beam model with multiple different models on a per key basis. This notebook uses pretrained pipelines pulled from Hugging Face. It is recommended that you walk through the [beginner RunInference notebook](https://colab.sandbox.google.com/github/apache/beam/blob/master/examples/notebooks/beam-ml/run_inference_pytorch_tensorflow_sklearn.ipynb) before continuing with this notebook."
+      ],
+      "metadata": {
+        "id": "ZAVOrrW2An1n"
+      }
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "## Install Dependencies\n",
+        "\n",
+        "We will first install Beam and some dependencies needed by Hugging Face"
+      ],
+      "metadata": {
+        "id": "_fNyheQoDgGt"
+      }
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 11,
+      "metadata": {
+        "colab": {
+          "base_uri": "https://localhost:8080/"
+        },
+        "id": "B-ENznuJqArA",
+        "outputId": "f72963fc-82db-4d0d-9225-07f6b501e256"
+      },
+      "outputs": [
+        {
+          "output_type": "stream",
+          "name": "stdout",
+          "text": [
+            ""
+          ]
+        }
+      ],
+      "source": [
+        "!pip install apache_beam[gcp]>=2.51.0 --quiet\n",
+        "!pip install torch --quiet\n",
+        "!pip install transformers --quiet\n",
+        "\n",
+        "# To use the newly installed versions, restart the runtime.\n",
+        "exit()"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "from typing import Dict\n",
+        "from typing import Iterable\n",
+        "from typing import Tuple\n",
+        "\n",
+        "from transformers import pipeline\n",
+        "\n",
+        "import apache_beam as beam\n",
+        "from apache_beam.ml.inference.base import KeyedModelHandler\n",
+        "from apache_beam.ml.inference.base import KeyModelMapping\n",
+        "from apache_beam.ml.inference.base import PredictionResult\n",
+        "from apache_beam.ml.inference.huggingface_inference import HuggingFacePipelineModelHandler\n",
+        "from apache_beam.ml.inference.base import RunInference"
+      ],
+      "metadata": {
+        "id": "wUmBEglvsOYW"
+      },
+      "execution_count": 1,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "## Define Configuration for our Models\n",
+        "\n",
+        "A `ModelHandler` is Beam's method for defining the configuration needed to load and invoke your model. Since we want to use multiple models, we will define 2 ModelHandlers, one for each model we're using in this example. Since both models being used are incapsulated by Hugging Face pipelines, we will use `HuggingFacePipelineModelHandler`.\n",
```

Comment:
```suggestion
        "A model handler is the Apache Beam method used to define the configuration needed to load and invoke models. Because this example uses two models, we define two model handlers, one for each model. Because both models are incapsulated within Hugging Face pipelines, we use the model handler `HuggingFacePipelineModelHandler`.\n",
```